langchain/prompts/few_shot.py

```diff
@@ -120,8 +140,73 @@ def _prompt_type(self) -> str:
         """Return the prompt type key."""
         return "few_shot"
 
-    def dict(self, **kwargs: Any) -> Dict:
-        """Return a dictionary of the prompt."""
-        if self.example_selector:
-            raise ValueError("Saving an example selector is not currently supported")
-        return super().dict(**kwargs)
+
+# For backwards compatibility.
+FewShotPromptTemplate = FewShotStringPromptTemplate
+
+
+class FewShotChatPromptTemplate(BaseChatPromptTemplate, BaseFewShotPromptTemplate):
+    """Prompt template that contains few shot examples."""
+
+    example_input: str
+    example_output: str
+    examples: List[Dict[str, str]]
+
+    @property
+    def _prompt_type(self) -> str:
+        """Return the prompt type key."""
+        return "few_shot_chat"
+
+    def format(self, **kwargs: Any) -> Sequence[BaseMessage]:
+        """Format to a sequence of BaseMessages."""
+
+        message_prompts = []
+        if self.prefix:
+            prefix_message_prompt = SystemMessagePromptTemplate(
+                prompt=StringPromptTemplate.from_template(self.prefix)
+            )
+            message_prompts.append(prefix_message_prompt)
+
+        # TODO: add support for example selectors
+        for example in self._get_examples():
```

Comment:
is this consistent with what open ai recommends? dont they have a name parameter?

Reply:
Yes, they do and I haven't supported that yet.
How about adding two more message types: `ExampleHumanMessage` and `ExampleAIMessage` -> In the `ChatOpenAI` class, we can translate these to OpenAI native messages by using the `name` parameter. For other models down the road that don't support `name`, we can just translate `ExampleHumanMessage` into a plain human message and same thing with `ExampleAIMessage`

Reply:
s/o to @Szpadel for suggesting this ^